pkg/util/kubernetes/hpa/datadogexternal.go

```diff
@@ -148,6 +149,11 @@ func NewDatadogClient() (*datadog.Client, error) {
 	if appKey == "" || apiKey == "" {
 		return nil, errors.New("missing the api/app key pair to query Datadog")
 	}
+
 	log.Infof("Initialized the Datadog Client for HPA")
-	return datadog.NewClient(apiKey, appKey), nil
+
+	client := datadog.NewClient(apiKey, appKey)
+	client.HttpClient.Transport = httputils.CreateHTTPTransport()
```

Comment:
Worth noting that this repackaging is the result of:
```
import cycle not allowed
package github.com/DataDog/datadog-agent/cmd/cluster-agent
	imports github.com/DataDog/datadog-agent/cmd/cluster-agent/app
	imports github.com/DataDog/datadog-agent/cmd/agent/common
	imports github.com/DataDog/datadog-agent/pkg/autodiscovery
	imports github.com/DataDog/datadog-agent/pkg/autodiscovery/configresolver
	imports github.com/DataDog/datadog-agent/pkg/autodiscovery/listeners
	imports github.com/DataDog/datadog-agent/pkg/util/kubernetes/apiserver
	imports github.com/DataDog/datadog-agent/pkg/util/kubernetes/hpa
	imports github.com/DataDog/datadog-agent/pkg/util
	imports github.com/DataDog/datadog-agent/pkg/util/hostname
	imports github.com/DataDog/datadog-agent/pkg/util/kubernetes/apiserver
```
when trying to use the import as is.